app/models/operatingsystems/xenserver.rb

```diff
@@ -0,0 +1,44 @@
+class Xenserver < Operatingsystem
+  PXEFILES = {:kernel => "boot/vmlinuz", :initrd => "install.img", :xen => "boot/xen.gz"}
+
+  def mediumpath(host)
+    medium_uri(host).to_s
+  end
+
+  def pxe_type
+    "xenserver"
+  end
+
+  def pxedir
+    ""
+  end
+
+  def xen(arch)
+    bootfile(arch,:xen)
+  end
+
+  def url_for_boot(file)
+    pxedir + "/" + PXEFILES[file]
+  end
+
+  def display_family
+    "XenServer"
+  end
+
+  def self.model_name
+    superclass.model_name
+  end
+
+  def bootfile(arch, type)
+    pxe_prefix(arch) + "-" + eval("#{self.family}::PXEFILES[:#{type}]").split("/")[-1]
+  end
+
+  def boot_files_uri(medium, architecture, host = nil)
```

Comment:
Could you add the URI of mboot.c32 also? I didn't try if it works, but that would open up the possibility to have all needed URLs for an iPXE template.